### DISCUSSION TITLE
(git.install) Terminate gpg-agent as part of install?

### DISCUSSION THREAD

**Discussion by brogers5** [chocolatey-community/chocolatey-packages#1924] [Ideas]
I originally raised this issue in the form of a [comment in #1451](https://github.com/chocolatey-community/chocolatey-packages/issues/1451#issuecomment-1159620838), but spinning off into a discussion per @TheCakeIsNaOH's request.

TL;DR: the `git.install` package has had a recurring issue with Git for Windows installations that have enabled signing commits with a GPG key. This spawns a background process (`gpg-agent.exe`) that must be manually closed by the user in order to upgrade the software. The package contains a `chocolateyBeforeModify.ps1` script to handle stopping this process, but this script only runs when `git.install` is upgraded directly. [If indirectly upgraded via the metapackage (i.e `git`), the script does not run](https://github.com/chocolatey/choco/issues/1092), and consequently the package fails to upgrade.

As a fix for the upstream issue does not appear to be planned for release soon, I'd like to propose we extract the `gpg-agent` termination code from `chocolateyBeforeModify.ps1`, encapsulate it within a function defined in `helpers.ps1`, and call the function from both `chocolateyBeforeModify.ps1` (to handle direct upgrades/uninstalls) and `chocolateyInstall.ps1` (as a workaround for indirect upgrades) to facilitate a smoother upgrade experience.

Thoughts?

**EDIT:** I'm also willing to own pull request creation if approved.

**Comment by AdmiringWorm** [comment]
This is something that should is considered a bug in Chocolatey CLI itself.
Fixing this issue should be done in Chocolatey CLI and not in the package itself.

However, as we currently do not have any ETA on when the upstream issue will be looked at, I'll approve and create a issue based on this discussion.
Since you mentioned you'd be willing to create a PR to enhance the package I'll assign the issue to you.

Just to note though, when adding the necessary code make sure that you add a comment in the install script of why it is needed to call the helper to stop the process.
